### PR TITLE
[GEOS-6790] Add support for GWC In-Memory caching for TileObjects

### PR DESCRIPTION
**Pull request by n-lagomarsini**
This pull request contains the fix for JIRA: http://jira.codehaus.org/browse/GEOS-6790. Note that this pull request requires to merge the related GWC counterpart (https://github.com/GeoWebCache/geowebcache/pull/295). More info about the GWC improvements can be found at the following link: https://github.com/GeoWebCache/geowebcache/issues/294.

### CODE REVIEW

**Comment by NielsCharlier** [comment]
Cannot be merged any more, conflicts must be resolved.

File: src/gwc/src/main/java/org/geoserver/gwc/config/GWCConfig.java [hunk 1]
```diff
@@ -31,6 +35,14 @@
     private boolean TMSEnabled;
     
     private boolean securityEnabled;
+    
+    private boolean innerCachingEnabled;
```
Comment (smithkm) [inline src/gwc/src/main/java/org/geoserver/gwc/config/GWCConfig.java]: Some comments, here or better yet as as javadocs on the accessor method methods to explain what "Inner Caching" and "Persistence" are would be helpful.

**Comment by n-lagomarsini** [comment]
Hi @smithkm,

Thank you for your review. I've added the javadocs and comments as you requested.

Regards,
Nicola.

**Comment by simboss** [comment]
I am going to move forward with the merge.

**Pull request merged**